client/dashboard/sites/site-fields.tsx

```diff
@@ -1,9 +1,41 @@
 import { useQuery } from '@tanstack/react-query';
 import { useInView } from 'react-intersection-observer';
+import { siteEngagementStatsQuery } from '../app/queries/site-stats';
 import { siteUptimeQuery } from '../app/queries/site-uptime';
 import { TextBlur } from '../components/text-blur';
 import type { Site } from '../data/types';
 
+export function EngagementStat( {
+	site,
+	type,
+}: {
+	site: Site;
+	type: 'visitors' | 'views' | 'likes';
+} ) {
+	const { ref, inView } = useInView( { triggerOnce: true, fallbackInView: true } );
+	const isEligible =
+		! site.is_deleted && ( ! site.jetpack || site.jetpack_modules?.includes( 'stats' ) );
```

Comment:
Looks like we need to think about how to reuse the eligibility check for the Jetpack feature 🤔

Reply:
Sorry for not being clear. I meant the check includes whether the deleted site is allowed and whether the non-jetpack site is allowed. However, I assume we can address it later.